src/components/Amount.vue

```diff
@@ -32,22 +43,19 @@ export default class Amount extends Vue {
             maxDecimals = this.maxDecimals;
         }
 
-        const roundingFactor = 10 ** maxDecimals;
-        const value = Math.floor((this.amount / 1e5) * roundingFactor) / roundingFactor;
+        const amount = moveComma(this.amount, -this.currencyDecimalCount);
+
+        const formattedNumber = formatNumber(amount, maxDecimals, minDecimals);
 
-        const result = parseFloat(value.toFixed(minDecimals)) === value
-            ? value.toFixed(minDecimals)
-            : value.toString();
+        if (formattedNumber !== amount) {
+            this.approximation = true;
+        }
```

Comment:
This approximation check is not correct as the formatted number can contain spaces and added decimals to reach `minDecimals`.
Also I don't like too much that the flag is set as a side effect of a getter.

I'd suggest to remove the flag and change `isApprox` to
```
const converted = moveComma(this.amount, -this.currencyDecimalCount);
return converted !== round(this.amount, this.decimals !== undefined ? this.decimals : this.maxDecimals);
```